core/src/action.rs

```diff
@@ -107,13 +107,13 @@ pub enum Action {
 
     /// Adds a link to the local DHT shard's meta/EAV storage
     /// Does not validate, assumes link is valid.
-    AddLink(Link),
+    AddLink(LinkData),
```

Comment:
I have actually been curious about our use of `LinkData` for add/remove actions. `LinkData` is just a `Link` plus a `LinkActionKind` which can be either `ADD` or `REMOVE`, isn't that redundant in the context of an action that's meant for adding links?

Reply:
LinkData contains AgentID and ISOTime right about now.

Reply:
Hmm not sure if we're talking about the same `LinkData`. I'm referring to this, which is just a `Link` with an enum attached. https://github.com/holochain/holochain-rust/blob/develop/core_types/src/link/link_data.rs#L13

Reply:
meant in this pull request ;) The LinkData will be used against the Entry::LinkAdd for generating a hash :)

You are right though, previous uses deem in redundant

Reply:
I still am not sure we're talking about the same thing. I am just wondering why we use `LinkData` instead of just `Link`. But my question isn't really necessary to answer for this PR, so I'll ask it elsewhere.

Reply:
OK I just got it. We are talking about the same thing. I didn't see that you added two fields to `LinkData`. But I am specifically wondering if we need `action_kind`. And that is a question for another day.